src/bundle/Resources/public/js/CKEditor/custom-attributes/custom-attributes-ui.js

```diff
@@ -0,0 +1,121 @@
+import Plugin from '@ckeditor/ckeditor5-core/src/plugin';
+import ButtonView from '@ckeditor/ckeditor5-ui/src/button/buttonview';
+import clickOutsideHandler from '@ckeditor/ckeditor5-ui/src/bindings/clickoutsidehandler';
+
+import IbexaCustomAttributesFormView from './ui/custom-attributes-form-view';
+
+class IbexaAttributesUI extends Plugin {
+    constructor(props) {
+        super(props);
+
+        this.balloon = this.editor.plugins.get('ContextualBalloon');
+        this.formView = this.createFormView();
+
+        this.showForm = this.showForm.bind(this);
+    }
+
+    getModelElement() {
+        return this.editor.model.document.selection.getSelectedElement() || this.editor.model.document.selection.anchor.parent;
+    }
+
+    createFormView() {
+        const formView = new IbexaCustomAttributesFormView({ locale: this.editor.locale });
+
+        this.listenTo(formView, 'save-custom-attributes', () => {
+            const values = this.formView.getValues();
+            const modelElement = this.getModelElement();
+
+            this.editor.model.change((writer) => {
+                Object.entries(values).forEach(([name, value]) => {
+                    writer.setAttribute(name, value, modelElement);
+                });
+            });
+
+            this.hideForm();
+        });
+
+        this.listenTo(formView, 'remove-custom-attributes', () => {
+            const values = this.formView.getValues();
+            const modelElement = this.getModelElement();
+
+            this.editor.model.change((writer) => {
+                Object.keys(values).forEach((name) => {
+                    writer.removeAttribute(name, modelElement);
+                });
+            });
+
+            this.hideForm();
+        });
+
+        return formView;
+    }
+
+    hideForm() {
+        this.balloon.remove(this.formView);
+        this.editor.editing.view.focus();
+    }
+
+    showForm() {
+        const parentElement = this.getModelElement();
+        const attributesValues = {};
+        const customAttributes = window.eZ.richText.alloyEditor.attributes[parentElement.name];
+        const customClasses = window.eZ.richText.alloyEditor.classes[parentElement.name];
+        const areCustomAttributesSet =
+            parentElement.hasAttribute('custom-classes') ||
+            Object.keys(customAttributes).some((customAttributeName) => parentElement.hasAttribute(customAttributeName));
+        let classesValue = areCustomAttributesSet ? parentElement.getAttribute('custom-classes') : customClasses.defaultValue;
+
+        Object.entries(customAttributes).forEach(([name, config]) => {
```

Comment:
use `.reduce` here instead of `forEach`?

```
const attributesValues = Object.entries(customAttributes).reduce((output, [name, config]) => {
  output[name] = areCustomAttributesSet ? parentElement.getAttribute(name) : config.defaultValue;
}, {});
```